### PR TITLE
Install yarn package manager for nodejs

### PR DESCRIPTION
**Pull request by ghn**
* This PR is a : New feature

- [x] Documentation is written
- [x] [Changelog](https://github.com/liip/drifter/blob/master/CHANGELOG.md) was updated

### CODE REVIEW

**Comment by LeBenLeBen** [comment]
In the idea of keeping the provisioning as fast as possible and since most people still use NPM as a default, I think this should be an option and defaults to false. Otherwise 👍

**Comment by ghn** [comment]
Good idea. I added the 'with_yarn' parameter. Default is False

File: provisioning/roles/nodejs/tasks/main.yml [hunk 1]
```diff
@@ -17,3 +17,18 @@
 - name: Install nodejs
   apt: name=nodejs state=present
   become: yes
+
+- name: Add Yarn apt key
+  apt_key: url=https://dl.yarnpkg.com/debian/pubkey.gpg
+  become: yes
+  when: with_yarn == true
```
Comment (sephii) [inline provisioning/roles/nodejs/tasks/main.yml]: Change to `when: with_yarn`

File: provisioning/roles/nodejs/tasks/main.yml [hunk 2]
```diff
@@ -17,3 +17,18 @@
...
+
+- name: Add Yarn apt repo
+  apt_repository: repo='deb https://dl.yarnpkg.com/debian/ stable main'
+  become: yes
+  when: with_yarn == true
```
Comment (sephii) [inline provisioning/roles/nodejs/tasks/main.yml]: Change to `when: with_yarn`

File: provisioning/roles/nodejs/tasks/main.yml [hunk 3]
```diff
@@ -17,3 +17,18 @@
...
+
+- name: Install yarn
+  apt: pkg=yarn state=latest
+  become: yes
+  when: with_yarn == true
```
Comment (sephii) [inline provisioning/roles/nodejs/tasks/main.yml]: Change to `when: with_yarn`

File: docs/roles/others.md [hunk 4]
```diff
@@ -8,12 +8,13 @@ or an unavailable package.
 
 # NodeJS
 
-Install NodeJS and NPM.
+Install NodeJS, NPM and Yarn.
```
Comment (sephii) [inline docs/roles/others.md]: Please add a note about Yarn not being installed by default. For example:

> Install NodeJS and NPM. Also install Yarn if `with_yarn` is true.

File: provisioning/roles/nodejs/defaults/main.yml [hunk 5]
```diff
@@ -8,3 +8,4 @@ nodejs_distro: "{{
 nodejs_acceptable_distros: ["wheezy", "jessie", "stretch", "sid", "precise", "trusty"]
 nodejs_version: "6.x"
 nodejs_acceptable_versions: ["7.x", "6.x", "5.x", "4.x", "0.12", "0.10"]
+with_yarn: false
```
Comment (sephii) [inline provisioning/roles/nodejs/defaults/main.yml]: Please prefix the parameter with the role name, eg. `nodejs_with_yarn`.

File: CHANGELOG.md [hunk 6]
```diff
@@ -10,6 +10,7 @@ For migration information, you can always have a look at https://liip-drifter.re
 
 ### Added
 - Node role: add support for version 7.x
+- Node role: install yarn
```
Comment (sephii) [inline CHANGELOG.md]: > - Node role: install yarn when `with_yarn` is set

**Comment by ghn** [comment]
@sephii fixed

**Comment by sephii** [comment]
Thanks! 🍟

**Pull request merged**